### PR TITLE
MM-1278 Team admin can now delete any post

### PR DESCRIPTION
**Pull request by nickago**

### CODE REVIEW

**Comment by rgarmsen2295** [comment]
Review complete, ready for merge

File: api/post.go [hunk 1]
```diff
@@ -619,16 +619,23 @@ func deletePost(c *Context, w http.ResponseWriter, r *http.Request) {
...
-		c.Err = result.Err
+	} else if presult := <-pchan; presult.Err != nil {
+		c.Err = presult.Err
 		return
 	} else {
-		post := result.Data.(*model.PostList).Posts[postId]
+
+		user := uresult.Data.(*model.User)
+
+		if !c.HasPermissionsToChannel(cchan, "deletePost") && !strings.Contains(user.Roles,"admin"){
```
Comment (coreyhulen) [inline api/post.go]: "admin" should be ROLE_ADMIN or ROLE_CHANNEL_ADMIN

Comment (coreyhulen) [inline api/post.go]: Also I think you could be an admin of a diff team or even a private channel.  I think you still need to check HasPermissionsToChannel (or some variation) vs skipping when admin role checked.

Comment (nickago) [inline api/post.go]: Alright, fix incoming

**Comment by nickago** [comment]
Ready for re-review

**Comment by rgarmsen2295** [comment]
Review complete, ready for merge

**Pull request merged**